Description: Add microsecond resolution to Message-id
 header to preserve the uniqueness
Author: Sergey B Kirpichev <skirpichev@gmail.com>
Bug-Debian: http://bugs.debian.org/613770

--- a/sendmail.c
+++ b/sendmail.c
@@ -66,6 +66,10 @@
 #include <string.h>
 #endif
 
+#ifdef HAVE_SYS_TIME_H
+#include <sys/time.h>
+#endif
+
 #include "monitor.h"
 #include "net.h"
 #include "socket.h"
@@ -190,7 +194,10 @@
     do_send(&S, "Mime-Version: 1.0\r\n");
     do_send(&S, "Content-Type: text/plain; charset=\"iso-8859-1\"\r\n");
     do_send(&S, "Content-Transfer-Encoding: 8bit\r\n");
-    do_send(&S, "Message-id: <%ld.%d@%s>\r\n", time(NULL), i, S.localhost);
+    struct timeval m_time;
+    gettimeofday(&m_time, NULL);
+    do_send(&S, "Message-id: <%ld.%ld,%d@%s>\r\n",
+            m_time.tv_sec, m_time.tv_usec, i, S.localhost);
     do_send(&S, "\r\n");
     do_send(&S, "%s\r\n", m->message);
     do_send(&S, ".\r\n");
